src/network/networkd-address.c

```diff
@@ -32,6 +32,8 @@
 #include "utf8.h"
 #include "util.h"
 
+#define STATIC_ADDRESSES_PER_NETWORK_MAX 1024U
```

Comment:
Trusted or not, I am pretty sure as a general rule we should enforce limits on everything (see other comment for a longer explanation).